Kudu.Services/Diagnostics/ProfileManager.cs

```diff
@@ -0,0 +1,316 @@
+﻿using Kudu.Contracts.Tracing;
+using Kudu.Core.Infrastructure;
+using Kudu.Core.Tracing;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Kudu.Services.Performance
+{
+    internal static class ProfileManager
+    {
+        private const int ProcessExitTimeoutInSeconds = 300;
```

Comment:
It's not really idle, if the CPU is at 100% on a small VM(single core) it will take a long time for the process to do its job and exit. I have been able to repro this and was taking around 120+ seconds. so I chose an upper limit. if azure has a 3 minute timeout I can change the code to 180 seconds then,